src/typing/type_inference_js.ml

```diff
@@ -982,7 +977,7 @@ and statement_decl cx = Ast.Statement.(
       let r = mk_reason (spf "module %s" name) loc in
       let t = Flow_js.mk_tvar cx r in
       Hashtbl.replace cx.type_table loc t;
-      Env_js.init_env cx (spf "$module__%s" name) (create_env_entry t t (Some loc))
+      Env_js.init_env cx (spf "$module__%s" name) (create_env_entry t t (Some loc)) VarScope (* TODO: hoist behavior *)
```

Comment:
I tried sticking module entries in Env_js.global_scope, but it doesn't work. I get `Required module not found` errors. Var scope seems to work fine, and the syntax prevents declaring modules in a nested scope, so I think it should be OK to leave declared modules as `VarScope`.